pkg/proxy/net/proxy.go

```diff
@@ -81,69 +85,72 @@ type Proxy struct {
 	Command    ProxyCommand
 }
 
-func (p *Proxy) ToBytes() []byte {
-	buf := make([]byte, len(proxyV2Magic)+4)
-	ptr := copy(buf, proxyV2Magic)
+func (p *Proxy) ToBytes() ([]byte, error) {
+	magicLen := len(proxyV2Magic)
+	buf := make([]byte, magicLen+4)
+	_ = copy(buf, proxyV2Magic)
+	buf[magicLen] = byte(p.Version<<4) | byte(p.Command&0xF)
```

Comment:
The command part should always be `0x01`, otherwise, the server will read the address of the proxy.
See https://github.com/blacktear23/go-proxyprotocol/blob/master/proxy_protocol.go#L311-L313

Reply:
That is controlled by the external part. This is just a correct library. And in the current implementation, it is always `PROXY`. proxy sould also handle `LOCAL`, but that will not be done in this PR.

Reply:
> If tiproxy forward LOCAL to tidb, it is also correct. That means another proxy wants to do health checking for TiDB through tiproxy.

Is it the only case when another proxy sends `LOCAL`? How did you get this information?

> Even if you rewrite the LOCAL to PROXY, you probably will get the address of the uppper proxy instead of a mysql client.

Why? Both TiProxy and TiDB will get the address of the upstream sender.

Reply:
> Is it the only case when another proxy sends LOCAL? How did you get this information?

The standard defines as so. If we want to enforce such behaviors, we could reject handshake for `LOCAL` command on tiproxy or tidb but still being responsive.

> Why? Both TiProxy and TiDB will get the address of the upstream sender.

Because `LOCAL` means the upstream sender is a proxy, instead of a client.

Reply:
```
LOCAL : the connection was established on purpose by the proxy
    without being relayed. 
PROXY : the connection was established on behalf of another node.
```

If a proxy sends `LOCAL` and TiProxy relays the message, then TiProxy should send `PROXY`, because it was established on behalf of another node.

Reply:
Makes sense, I could do the change. I think the core point is that `LOCAL` does not care about the original IP, but anyway.